lib/executor/helpers.go

```diff
@@ -128,7 +128,7 @@ func getIterationRunner(
 					// TODO don't count this as a full iteration?
 					logger.WithField("source", "stacktrace").Error(exception.StackTrace())
 				} else {
-					logger.Error(err.Error())
+					logger.WithField("source", "throwable").Error(err.Error())
```

Comment:
I would argue that this just hides the underlying problem.

The reported issue is that an asynchronous exception isn't reported as one. 

You here make any error that isn't an exception to have a `source` "throwable" ... which I would argue is not true even for the example you have and will likely just hide any future problem.

The solution to the linked issue should be done in the eventloop where the error is returned and it should be come an exception. If possible I guess it should rewrap the original one ... somehow.

https://github.com/grafana/k6/blob/c5dc773703e8f44ebce81ab929b7dd741b803335/js/eventloop/eventloop.go#L184-L198

Reply:
hi @mstoykov, thanks for the feedback. I realize my changes don't fully address the issue and might cause more problems. I'm not sure how to proceed with the solution you've suggested, so I'll close this PR for now. 
Again, thanks for your time and guidance!